numba_cuda/numba/cuda/tests/cudapy/test_bfloat16_bindings.py

```diff
@@ -22,18 +22,17 @@
     hexp,
     hexp2,
     hexp10,
+    htanh,
+    htanh_approx,
 )
-
-from numba.cuda.cudadrv.runtime import get_version
-
-cuda_version = get_version()
+from numba.cuda import is_bfloat16_supported
 
 dtypes = [int16, int32, int64, uint16, uint32, uint64, float32]
 
 
 @unittest.skipIf(
-    (cuda.get_current_device().compute_capability < (8, 0)),
-    "bfloat16 requires compute capability 8.0+",
+    not is_bfloat16_supported(),
```

Comment:
I think this is going to call `runtime.get_version()` at test discovery time - it might be working OK but I think it would be safer to do:

```python
if not is_bfloat16_supported():
    self.skipTest()
```

inside the test body. (It's not an ideal pattern, but there've been other instances where I've had to do this in the test suite).

Reply:
(I just noticed this was already not a good pattern because of calling `cuda.get_current_device()` at test dicsovery time - I missed that in previous reviews.

Reply:
Ah - would you say the implicit rule is that the device should be initialized right before the first time we actually invoke a kernel?

Reply:
It should be initialized right before the first time we need to interact with the device - it needn't be a kernel launch - it could be an allocation, an enumeration of the devices in the machine (i.e. `cuda.detect()`), etc.

I think I was thinking of the rule as "not at import time, and not at test discovery time", but your rule probably puts a closer bound on it.